vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -636,11 +682,35 @@ public void setRenderer(
         refreshCheckboxItems();
     }
 
-    @SuppressWarnings("unchecked")
-    private void reset() {
-        keyMapper.removeAll();
-        clear();
+    /**
+     * Sets the selection preservation strategy on data change. The default is
```

Comment:
"Data change" is maybe a bit too vague, it's not clear that this is effectively related to `DataProvider.refreshAll`. Maybe something like:
```
Determines what happens with the selection when the data provider is refreshed.
```
Or:
```
Determines what happens with the selection when the {@link DataProvider#refreshAll} is called.
```

Another useful note could be that the selection is discarded in any case when a new data provider is set.

Reply:
Updated the JavaDocs to mention `DataProvider.refreshAll` relation and discarding selection on setting data provider.